File: products/amliq/api/runlocal/internal/middleware/cors.go
package middleware

import (
	"net/http"
)

var allowedOrigins = map[string]bool{
	"https://pushci.dev":              true,
	"https://pushci-app.pages.dev":    true,
	"http://localhost:5173":           true,
	"http://localhost:3000":           true,
}

const (
	corsMaxAge       = "86400"
	corsMethods      = "GET, POST, PUT, DELETE, OPTIONS"
	corsHeaders      = "Authorization, Content-Type"
)

// CORS returns middleware that sets CORS headers.
func CORS(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		origin := r.Header.Get("Origin")
		if allowedOrigins[origin] {
			w.Header().Set("Access-Control-Allow-Origin", origin)
			w.Header().Set("Access-Control-Allow-Methods", corsMethods)
			w.Header().Set("Access-Control-Allow-Headers", corsHeaders)
			w.Header().Set("Access-Control-Max-Age", corsMaxAge)
		}

		if r.Method == http.MethodOptions {
			w.WriteHeader(http.StatusNoContent)
			return
		}

		next.ServeHTTP(w, r)
	})
}

File: products/queryflux/backend/server-ts/middleware/cors.ts
import type { Request, Response, NextFunction } from 'express';

export function corsMiddleware(allowedOrigins: string[]) {
  return (req: Request, res: Response, next: NextFunction): void => {
    const origin = req.headers.origin || '';

    if (allowedOrigins.includes(origin) || allowedOrigins.includes('*')) {
      res.setHeader('Access-Control-Allow-Origin', origin);
    }

    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');
    res.setHeader('Access-Control-Max-Age', '86400');
    res.setHeader('Access-Control-Allow-Credentials', 'true');

    if (req.method === 'OPTIONS') {
      res.status(204).end();
      return;
    }

    next();
  };
}
